core/controlplane/config/config.go

```diff
@@ -724,50 +732,51 @@ func (c Cluster) StackConfig(opts StackTemplateOptions) (*StackConfig, error) {
 	var err error
 	stackConfig := StackConfig{}
 
+	var compactAssets *CompactTLSAssets
+	var compactAuthTokens *CompactAuthTokens
+
 	if stackConfig.Config, err = c.Config(); err != nil {
 		return nil, err
 	}
 
-	// TODO: Check if new tests are needed to verify the auth token file is handled correctly
+	if c.AssetsEncryptionEnabled() {
+		compactAuthTokens, err = ReadOrCreateCompactAuthTokens(opts.AssetsDir, KMSConfig{
+			Region:         stackConfig.Config.Region,
+			KMSKeyARN:      c.KMSKeyARN,
+			EncryptService: c.ProvidedEncryptService,
+		})
+		if err != nil {
+			return nil, err
+		}
+		stackConfig.Config.AuthTokensConfig = compactAuthTokens
+	} else {
+		rawAuthTokens, err := ReadOrCreateUnencryptedCompactAuthTokens(opts.AssetsDir)
+		if err != nil {
+			return nil, err
+		}
+		stackConfig.Config.AuthTokensConfig = rawAuthTokens
+	}
```

Comment:
@mumoshu Yes, I found and fixed the bug here before #470 was opened. If it gets merged before this PR, I'll remove this.